stdlib/LinearAlgebra/src/generic.jl

```diff
@@ -33,18 +33,22 @@ struct MulAddMul{ais1, bis0, TA, TB}
     beta::TB
 end
 
-@inline function MulAddMul(alpha::TA, beta::TB) where {TA,TB}
+function MulAddMul{ais1,bis0}(alpha::TA, beta::TB) where {ais1,bis0,TA,TB}
+    MulAddMul{ais1,bis0,TA,TB}(alpha,beta)
```

Comment:
Looks like you can create inconsistent `MulAddMul` objects then? Isn't that a little bit dangerous? If somebody has a typo in the alpha, beta arguments, but uses the original `ais1` and `bis0`?

Reply:
I've added these checks now. Unfortunately, I am not being able to compile away the checks using constant-propagation.